interfaces/kmod/backend.go

```diff
@@ -32,11 +53,65 @@ func (b *Backend) Name() string {
 	return "kmod"
 }
 
+// Setup creates a conf file with list of kernel modules required by given snap,
+// writes it in /etc/modules-load.d/ directory and immediately loads the modules
+// using /sbin/modprobe. The devMode is ignored.
+//
+// If the method fails it should be re-tried (with a sensible strategy) by the caller.
 func (b *Backend) Setup(snapInfo *snap.Info, devMode bool, repo *interfaces.Repository) error {
-	// TODO: get snippets, load modules, create /etc/modules-load.d/snap.modules.conf file
-	return nil
+	snapName := snapInfo.Name()
+	// Get the snippets that apply to this snap
+	snippets, err := repo.SecuritySnippetsForSnap(snapInfo.Name(), interfaces.SecurityKMod)
+	if err != nil {
+		return fmt.Errorf("cannot obtain kmod security snippets for snap %q: %s", snapName, err)
+	}
```

Comment:
You can use `osutil.EnsureDirState` for this. All the other backends do it and it is nicely testable.